virtcontainers/qemu.go

```diff
@@ -1095,3 +1098,25 @@ func genericMemoryTopology(memoryMb, hostMemoryMb uint64) govmmQemu.Memory {
 
 	return memory
 }
+
+func (q *qemu) hotplugSocket(socket Socket, op operation) error {
+	if err := q.qmpSetup(); err != nil {
+		return err
+	}
+
+	if op == addDevice {
+		err := q.qmpMonitorCh.qmp.ExecuteCharDevUnixSocketAdd(q.qmpMonitorCh.ctx, socket.ID, socket.HostPath, false, true)
+		if err != nil {
+			return err
+		}
+
+		err = q.qmpMonitorCh.qmp.ExecuteVirtSerialPortAdd(q.qmpMonitorCh.ctx, socket.DeviceID, socket.Name, socket.ID)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New("Remove socket device is not supported")
```

Comment:
You're making the assumption that `operation` has only two types here. If we get a third possible `operation`, we might want to throw a more generic error here with the stringified name of the operation that's not supported.
Also, we could decide that we won't get more than 2 operations, and in this case, we should use a boolean.